core/Pointer.carp

```diff
@@ -20,6 +20,28 @@ The user will have to ensure themselves that this is a safe operation.")
   (doc from-long "converts a long integer to a pointer.")
   (deftemplate from-long (Fn [Long] (Ptr p)) "$p* $NAME(Long p)" " $DECL { return ($p*)p; }")
 
+  (doc unsafe-set
+    "Sets the value of a pointer to a value of any type."
+    "The user will have to ensure this operation is safe.")
+  (deftemplate unsafe-set (Fn [(Ptr p) a] Unit) "void $NAME($p* p, $a a)" "$DECL { *p = ($p)a; }")
+
+  (doc unsafe-alloc
+     "Allocate a new pointer to the value `a`."
+     ("This pointer won't be managed by Carp's borrow checker and will cause leaks " false)
+     "unless explicitly freed."
+     ""
+     ("See `Unsafe.leak` if you want to prevent Carp's automatic destructuring " false)
+     "on a value without performing an allocation.")
+  (deftemplate unsafe-alloc (Fn [a] (Ptr a)) "$a* $NAME($a r)" "$DECL { void *leak = CARP_MALLOC(sizeof($a)); *($a*)leak= r; return ($a*)leak;}")
+
+  (doc free
+     "Free a pointer."
```

Comment:
Wonder if it's worth being a bit clearer what free does to non-C Programmers...

Reply:
Will do!